From 1db88d904f124636c7ade8fdd7881d0fd2d6caf1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Martinez?= <me@moverest.xyz>
Date: Mon, 14 Apr 2025 06:48:08 +0100
Subject: [PATCH] Don't enter first mode if layer surface not configured

On some compositors (Niri) the `surface.enter()` is called before the
layer surface configure event is sent. This means that we don't have the
surface's dimensions, we try to enter the mode anyway, and we break.
This change prevents that.
---
 src/main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index f7bb237..81a5886 100644
--- a/src/main.c
+++ b/src/main.c
@@ -439,7 +439,9 @@ static void handle_surface_enter(
         find_output_from_wl_output(&state->outputs, wl_output);
     state->current_output = output;
 
-    enter_first_mode(state);
+    if (state->surface_configured) {
+        enter_first_mode(state);
+    }
 }
 
 static const struct wl_surface_listener surface_listener = {
